ocaml/testsuite/tests/typing-layouts/basics_alpha.ml

```diff
@@ -1113,12 +1110,11 @@ let f20 () =
   in
   ();;
 [%%expect{|
-Lines 4-5, characters 4-5:
-4 | ....let module M = struct end in
-5 |     x
-Error: Non-value detected in translation:
+Line 3, characters 6-8:
+3 |   let _y =
```

Comment:
I think this just happens because `translcore` translates the body of a let before the bound term.  But we do reject the thing you are worried about - it's rejected by the type system if at the module level, and by the safety check otherwise:

```ocaml
# type t_void [@@void];;
type t_void [@@void]

# let x : t_void = assert false;;
Error: Top-level module bindings must have layout value, but x has layout
       void.
       
# let foo : int =
    let x : t_void = assert false in
    42;;
Warning 26 [unused-var]: unused variable x.
Error: Non-value detected in [Typeopt.layout] as sort for type t_void.
       Please report this error to the Jane Street compilers team.
```

(Perhaps we'll want to relax the top-level binding rule to "not void" in a subsequent layouts PR.)